pdns-recursor from GIT : compilation error on opensslsigners

on fedora 14/17 with latest gcc:

```
  CXX    gettime.o
  CXX    gss_context.o
  CXX    iputils.o
  CXX    ixfr.o
  CXX    json.o
  CXX    lua-recursor4.o
  CXX    lwres.o
  CXX    mtasker_context.o
  CXX    nsecrecords.o
  CXX    opensslsigners.o
opensslsigners.cc:131:41: erreur : 'ECDSA_SIG' does not name a type
 static inline void ECDSA_SIG_get0(const ECDSA_SIG* signature, const BIGNUM** pr, const BIGNUM** ps) {
                                         ^~~~~~~~~
opensslsigners.cc: Dans la fonction 'void ECDSA_SIG_get0(const int*, const BIGNUM**, const BIGNUM**)':
opensslsigners.cc:132:20: erreur : request for member 'r' in '* signature', which is of non-class type 'const int'
   *pr = signature->r;
                    ^
opensslsigners.cc:133:20: erreur : request for member 's' in '* signature', which is of non-class type 'const int'
   *ps = signature->s;
                    ^
opensslsigners.cc: At global scope:
opensslsigners.cc:136:34: erreur : 'ECDSA_SIG_set0' declared as an 'inline' variable
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                  ^~~~~~~~~
opensslsigners.cc:136:34: erreur : 'ECDSA_SIG' was not declared in this scope
opensslsigners.cc:136:45: erreur : 'signature' was not declared in this scope
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                             ^~~~~~~~~
opensslsigners.cc:136:62: erreur : expected primary-expression before '*' token
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                              ^
opensslsigners.cc:136:64: erreur : 'pr' was not declared in this scope
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                ^~
opensslsigners.cc:136:74: erreur : expected primary-expression before '*' token
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                          ^
opensslsigners.cc:136:76: erreur : 'ps' was not declared in this scope
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                            ^~
opensslsigners.cc:136:78: erreur : expression list treated as compound expression in initializer [-fpermissive]
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                                                                              ^
opensslsigners.cc:136:19: attention : 'ECDSA_SIG_set0' defined but not used [-Wunused-variable]
 static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
                   ^~~~~~~~~~~~~~
make[2]: *** [opensslsigners.o] Erreur 1
make[2] : on quitte le répertoire « /root/pascal/pdns_recursor_git/pdns-master/pdns/recursordist »
make[1]: *** [all-recursive] Erreur 1
make[1] : on quitte le répertoire « /root/pascal/pdns_recursor_git/pdns-master/pdns/recursordist »
make: *** [all] Erreur 2
```

--- a/m4/pdns_check_libcrypto_ecdsa.m4
+++ b/m4/pdns_check_libcrypto_ecdsa.m4
@@ -12,6 +12,6 @@ AC_DEFUN([PDNS_CHECK_LIBCRYPTO_ECDSA], [
   ])
 
   AS_IF([test "x$libcrypto_ecdsa" = "xyes"], [
-    AC_DEFINE([HAVE_LIBCRYPTO_ECDSA], [1], [define to 1 if OpenSSL ecdsa support is avalable.])
+    AC_DEFINE([HAVE_LIBCRYPTO_ECDSA], [1], [define to 1 if OpenSSL ecdsa support is available.])
   ])
 ])

--- a/pdns/opensslsigners.cc
+++ b/pdns/opensslsigners.cc
@@ -128,6 +128,7 @@ static inline int RSA_set0_crt_params(RSA* rsakey, BIGNUM* dmp1, BIGNUM* dmq1, B
   return 1;
 }
 
+#ifdef HAVE_LIBCRYPTO_ECDSA
 static inline void ECDSA_SIG_get0(const ECDSA_SIG* signature, const BIGNUM** pr, const BIGNUM** ps) {
   *pr = signature->r;
   *ps = signature->s;
@@ -140,6 +141,8 @@ static inline int ECDSA_SIG_set0(ECDSA_SIG* signature, BIGNUM* pr, BIGNUM* ps) {
   signature->s = ps;
   return 1;
 }
+#endif /* HAVE_LIBCRYPTO_ECDSA */
+
 #else
 void openssl_thread_setup() {}
 void openssl_thread_cleanup() {}

